manager/supermanager.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2021 VMware, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package manager
+
+import (
+	"context"
+	"math/big"
+	"sync"
+
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+)
+
+// New returns a controller manager which can subquently create other managers
+// with the given REST configuration and controller options. If the REST
+// configuration or anything it refers to is mutated or if anything the
+// controller options refers to is subsequently mutated, unpredictable behaviour
+// may occur.
+func New(config *rest.Config, options ctrl.Options) (SuperManager, error) {
+	mainManager, err := ctrl.NewManager(config, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return &superManager{
+		Manager:     mainManager,
+		config:      config,
+		options:     options,
+		failures:    make(chan error),
+		nextContext: big.NewInt(0),
+		contexts:    make(map[*big.Int]context.CancelFunc),
+	}, nil
+}
+
+// SuperManager is a controller manager which can create other managers in order
+// to be able to stop watches by stopping the corresponding caches.
+type SuperManager interface {
+	ctrl.Manager
+
+	// NewManager returns a new controller manager which is configured the same
+	// as this super manager.
+	NewManager() (ctrl.Manager, error)
```

Comment:
Not yet. The idea is that ACP, or another user, will call this in `main.go` instead of calling `ctrl.NewManager` (i.e. the corresponding API in controller runtime).

Reply:
Good catch! There is some missing code to actually create managers that can be cancelled to stop watchers.

Reply:
Missing code added to second commit.